### PR TITLE
Finalize hwmv2 updates for Infineon devices

### PR DESCRIPTION
**Pull request by npal-cy**
Finalize hwmv2 updates for Infineon devices

### CODE REVIEW

**Review by ifyall — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]
Not reviewed changes but temp. block until https://github.com/zephyrproject-rtos/zephyr/pull/70548 has been merged

**Comment by nordicjm** [comment]
Needs rebase

**Comment by npal-cy** [comment]
> Not reviewed changes but temp. block until #70548 has been merged

Hi @nordicjm  Do we still waiting #70548? or it handle in alternative https://github.com/zephyrproject-rtos/zephyr/pull/71149.

**Comment by nordicjm** [comment]
> > Not reviewed changes but temp. block until #70548 has been merged
> 
> Hi @nordicjm Do we still waiting #70548? or it handle in alternative #71149.

That issue has been resolved

File: soc/infineon/cat1a/Kconfig [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Category definitions
+config SOC_FAMILY_INFINEON_CAT1
+	bool
+
+config SOC_FAMILY_INFINEON_CAT1A
+	bool
+
+# Family definitions
+config SOC_FAMILY_PSOC6
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig:15]: This is not correct, revert to how it was done previously

File: soc/infineon/cat1a/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# PSOC CAT1B Configuration
+
+# Copyright (c) 2024 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+rsource "*/Kconfig.defconfig"
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig.defconfig:6]: likewise here, probably other places too, revert

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/infineon/cat1a/Kconfig (hunk 1 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig]: updated

File: soc/infineon/cat1a/Kconfig.defconfig (hunk 2 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig.defconfig:6]: updated

File: soc/infineon/cat1a/Kconfig.soc [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+config SOC_PSOC6_M0_ENABLES_M4
+	bool "Dual-core support [activate Cortex-M4]"
+	depends on SOC_FAMILY_PSOC6_LEGACY_M0
+	help
+	  Cortex-M0 CPU should boot Cortex-M4
+
+# Cypress PSoC™ 6 MCU lines
+config SOC_SERIES_PSOC_60
+	bool "Infineon PSoC™ 60 series MCU (Value Line)"
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig.soc:53]: no selections for SOC_SERIES, they are selected automatically by the build system

File: soc/infineon/cat1a/Kconfig [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Copyright (c) 2024 Cypress Semiconductor Corporation (an Infineon company) or
+# an affiliate of Cypress Semiconductor Corporation
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig:2]: empty file can go though I don't see why this file is empty?

File: soc/infineon/cat1a/Kconfig.soc [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	bool
+
+# Family definitions
+config SOC_FAMILY_PSOC6
+	bool
+	select ARM
+	select CPU_CORTEX_M4
+	select CPU_HAS_ARM_MPU  
+	select DYNAMIC_INTERRUPTS
+	select CPU_HAS_FPU
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig.soc:24]: this is incorrect, look at how other boards are. SOC_* symbols are selected in Kconfig.soc, anything else is selected in Kconfig

File: soc/infineon/cat1a/Kconfig.soc [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	help
+	  Enable support for Infineon PSoC™ 64 MCU series
+
+config SOC_SERIES
+	default "psoc6_60" if SOC_SERIES_PSOC_60
+	default "psoc6_61" if SOC_SERIES_PSOC_61
+	default "psoc6_62" if SOC_SERIES_PSOC_62
+	default "psoc6_63" if SOC_SERIES_PSOC_63
+	default "psoc6_64" if SOC_SERIES_PSOC_64
+
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig.soc:93]: everything here needs to match so if you have `default "psoc6_61"` then you need to have `SOC_SERIES_PSOC6_61`

File: soc/infineon/cat1a/Kconfig.soc [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	config SOC_PSOC6_CM0P_IMAGE_ROM_SIZE
+	hex
+	default 0x2000 if SOC_PSOC6_CM0P_IMAGE_SLEEP
+
+	config SOC_PSOC6_CM0P_IMAGE_RAM_SIZE
+	hex
+	default 0x2000 if SOC_PSOC6_CM0P_IMAGE_SLEEP
+
+endif # SOC_FAMILY_PSOC6
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig.soc:119]: indent completely wrong, goes in Kconfig

File: soc/infineon/cat1a/psoc6_legacy/Kconfig.soc [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 Cypress Semiconductor Corporation (an Infineon company) or
+# an affiliate of Cypress Semiconductor Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Infineon PSoC6 (legacy) series MCUs
+config SOC_CY8C6247_M0
+	bool
+	select SOC_SERIES_PSOC_62
```
Comment (nordicjm) [inline soc/infineon/cat1a/psoc6_legacy/Kconfig.soc:8]: this file is OK, use this as a base

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/infineon/cat1a/Kconfig.soc (hunk 7 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig.soc]: updated. moved to Kconfig (from Kconfig.soc), removed indent.

File: soc/infineon/cat1a/Kconfig.soc (hunk 6 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig.soc]: done

File: soc/infineon/cat1a/Kconfig (hunk 4 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig:2]: now it is not empty.

File: soc/infineon/cat1a/Kconfig.soc (hunk 3 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig.soc]: removed "...."

File: soc/infineon/cat1a/Kconfig.soc (hunk 5 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig.soc]: updated

**Comment by npal-cy** [comment]
can we merge this PP? there are another PRs  blocked by this one.

File: modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt [hunk 9]
```diff
@@ -19,18 +19,21 @@ zephyr_library_sources(${pdl_dev_cat1a_dir}/source/cy_device.c)
...
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_PWM      ${pdl_drv_dir}/source/cy_tcpwm_pwm.c)
+#zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_RTC      ${pdl_drv_dir}/source/cy_rtc.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_SDIO     ${pdl_drv_dir}/source/cy_sd_host.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_SPI      ${pdl_drv_dir}/source/cy_scb_spi.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_TIMER    ${pdl_drv_dir}/source/cy_tcpwm_counter.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_UART     ${pdl_drv_dir}/source/cy_scb_uart.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_FLASH    ${pdl_drv_dir}/source/cy_flash.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_WDT      ${pdl_drv_dir}/source/cy_wdt.c)
 
+zephyr_library_sources(${pdl_drv_dir}/source/cy_tcpwm_pwm.c)
```
Comment (nordicjm) [inline modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt:35]: why is this not guarded with an ifdef?

File: modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt [hunk 10]
```diff
@@ -19,18 +19,21 @@ zephyr_library_sources(${pdl_dev_cat1a_dir}/source/cy_device.c)
 zephyr_library_sources(${pdl_drv_dir}/source/TOOLCHAIN_GCC_ARM/cy_syslib_ext.S)
 
 # Peripheral drivers
-zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6      ${pdl_drv_dir}/source/cy_sysint.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6_LEGACY      ${pdl_drv_dir}/source/cy_sysint.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_ADC      ${pdl_drv_dir}/source/cy_sar.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_I2C      ${pdl_drv_dir}/source/cy_scb_i2c.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_LPTIMER  ${pdl_drv_dir}/source/cy_mcwdt.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_PWM      ${pdl_drv_dir}/source/cy_tcpwm_pwm.c)
+#zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_RTC      ${pdl_drv_dir}/source/cy_rtc.c)
```
Comment (nordicjm) [inline modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt:27]: why is this commented?

File: modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt [hunk 11]
```diff
@@ -47,6 +50,7 @@ if(CONFIG_USE_INFINEON_ADC)
   zephyr_library_sources(${pdl_drv_dir}/source/cy_sysanalog.c)
 endif()
 
+
```
Comment (nordicjm) [inline modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt:53]:
```suggestion
```

File: soc/infineon/cat1a/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  # Add section for cm0p image ROM
+  zephyr_linker_sources(ROM_START SORT_KEY 0x0cm0p common/rom_cm0image.ld)
+
+  # Add section for cm0p image RAM
+  zephyr_linker_sources(RAM_SECTIONS SORT_KEY 0 common/ram_cm0image.ld)
+  zephyr_linker_sources(RAMFUNC_SECTION SORT_KEY 0 common/ram_func.ld)
+  zephyr_linker_sources(RODATA SORT_KEY 0 common/rom.ld)
+
+  zephyr_compile_definitions_ifdef(CONFIG_SOC_CYBLE_416045_02_DEVICE CYBLE_416045_02_device)
+
```
Comment (nordicjm) [inline soc/infineon/cat1a/CMakeLists.txt:20]:
```suggestion
```

File: soc/infineon/cat1a/Kconfig.defconfig (hunk 2 above)
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig.defconfig:6]: looks like this is missing a SOC_SERIES_* guard?

File: soc/infineon/cat1a/Kconfig.soc [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+	select SOC_FAMILY_INFINEON_CAT1A
+	help
+	  Enable support for Infineon PSoC™ 63 MCU series
+
+config SOC_SERIES_PSOC6_64
+	bool
+	select SOC_FAMILY_PSOC6 if !SOC_FAMILY_PSOC6_LEGACY
+	select SOC_FAMILY_INFINEON_CAT1A
+	help
+	  Enable support for Infineon PSoC™ 64 MCU series
```
Comment (nordicjm) [inline soc/infineon/cat1a/Kconfig.soc:64]: help text for all of these can go

File: soc/infineon/cat1a/soc.yml [hunk 14]
```diff
@@ -129,3 +114,12 @@ family:
     - name: cy8c6144lqi_s4f62
     - name: cy8c6144azi_s4f12
     - name: cy8c6144lqi_s4f12
+    - name: cyble_416045_02_device
```
Comment (nordicjm) [inline soc/infineon/cat1a/soc.yml:117]: why does this now have device in the name?

Comment (nordicjm) [reply soc/infineon/cat1a/soc.yml:117]: I had a quick search for this and can't find any such soc on your website?

File: modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt (hunk 10 above)
Comment (npal-cy) [reply modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt]: reverted

File: soc/infineon/cat1a/CMakeLists.txt (hunk 12 above)
Comment (npal-cy) [reply soc/infineon/cat1a/CMakeLists.txt]: removed

File: soc/infineon/cat1a/Kconfig.defconfig (hunk 2 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig.defconfig:8]: added

File: soc/infineon/cat1a/soc.yml (hunk 14 above)
Comment (npal-cy) [reply soc/infineon/cat1a/soc.yml]: cyble_416045_02 this is module based on PSoC 63 https://www.infineon.com/cms/en/product/wireless-connectivity/airoc-bluetooth-le-bluetooth-multiprotocol/airoc-bluetooth-modules/cyble-416045-02/

In PDL, modules has "_device" ending in device headers:   https://github.com/Infineon/mtb-pdl-cat1/blob/master/devices/COMPONENT_CAT1A/include/cyble_416045_02_device.h

To user prevent confusing i changed :  cyble_416045_02_device -> cyble_416045_02

in [modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt](https://github.com/zephyrproject-rtos/zephyr/compare/ffacb3614304cf6411d00a25fef9d1d32ddeca8d..324a0fb218b441c4ad8778b4325515d286927ef4#diff-c4402b261bf1d521423ef6072e0640a118e7c655828defc67eae86f0a99c6a78) i added  zephyr_compile_definitions_ifdef for CONFIG_SOC_CYBLE_416045_02  to handling PDL related devines..
```
# Generate PDL specific define (w. *_device) for SOC module (e.g: CYBLE_416045_02) 
zephyr_compile_definitions_ifdef(CONFIG_SOC_CYBLE_416045_02 CYBLE_416045_02_device)
```

File: soc/infineon/cat1a/Kconfig.soc (hunk 13 above)
Comment (npal-cy) [reply soc/infineon/cat1a/Kconfig.soc]: removed

**Comment by npal-cy** [comment]
can we merge this one? This PR block another PRs.

**Review by ifyall — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**